developer-docs/latest/concepts/sso.md

```diff
@@ -0,0 +1,293 @@
+| `icon`           | false    | string   | An image URL. If specified, it will replace the displayName on the login page  |
+| `createStrategy` | true     | function | A factory that will build and return a new passport strategy for your provider. Takes the strapi instance as parameter |
+
+::: tip
+The `uid` property is the unique identifier of each strategy and is generally found in the strategy's package.
+
+If you are not sure of what it refers to, please contact the maintainer of the strategy.
+:::
+
+#### The `createStrategy` Factory
+
+A passport strategy is usually built by instantiating it using two parameters:
+
+##### Configuration Object
+
+The configuration object depends on the strategy needs, but often ask for a callback url to be redirected to once the connection has been made on the provider side.
+
+You can generate a specific callback url for your provider using the `getProviderCallbackUrl` method.
+
+::: tip
+`strapi.admin.services.passport.getProviderCallbackUrl` is a Strapi util you can use to get a callback URL for a specific provider. It takes a provider name as a parameter and returns a URL as a string.
+:::
+
+If needed, this is also where you will put your client ID and secret key for your OAuth2 application.
+
+##### Verify Function
+
+The verify function is used here as a middleware allowing the user to transform and make extra processing on the data returned from the provider API.
+
+This function always takes a `done` method as last parameter which is used to transfer needed data to the Strapi layer of SSO.
+
+Its signature is the following: `void done(error: any, data: object);` and it follows the following rules:
+
+- If `error` is not set to `null`, then the data sent is ignored, and the controller will throw an error.
+- If the SSO's auto-registration feature is disabled, then the `data` object only need to be composed of an `email` property.
+- If the SSO's auto-registration feature is enabled, then you will need to define (in addition to the `email`) either a `username` property or both `firstname` and `lastname` within the `data` oject.
+
+### Adding a provider
+
+Adding a new provider means adding a new way for your administrators to log-in.
+
+To achieve a great flexibility and a large choice of provider, we use [Passport.js](http://www.passportjs.org/) under the hood. It means that any valid passport strategy that doesn't need additional custom data should also work with Strapi. 
+
+::: warning
+Strategies such as [ldapauth](https://github.com/vesse/passport-ldapauth) don't work out of the box since they require extra data to be sent from the admin panel.
+If you want to add an LDAP provider to your application, you will need to write a [custom strategy](http://www.passportjs.org/packages/passport-custom/).
+You can also use services such as Okta and Auth0 as bridge services.
+:::
+
+### Configuring the provider
+
+Configuring a new provider is straight-forward:
```

Comment:
As we should avoid using the word "simple", we should avoid using similar words such as "straight-forward".
Instead, you could introduce the procedure by something like "To configure a new provider[, follow the procedure below]:".